### PR TITLE
[DOCS] Create modular example

### LINKED ISSUE #4636: Give a modular example for an ERC 20 implementation in the Solidity by example section
We should check if that implementation really is in line with the recent security recommendations for ERC20 implementations. Perhaps we should even implement one of the newer standards.

```
library Balances {
    function send(mapping(address => uint256) storage balances, address from, address to, uint amount) internal {
        require(balances[from] >= amount);
        require(balances[to] + amount >= balances[to]);
        balances[from] -= amount;
        balances[to] += amount;
    }
}

contract Token {
    mapping(address => uint256) balances;
    using Balances for *;
    mapping(address => mapping (address => uint256)) allowed;

    event Transfer(address from, address to, uint amount);
    event Approval(address owner, address spender, uint amount);

    function balanceOf(address tokenOwner) public constant returns (uint balance) {
        return balances[tokenOwner];
    }
    function transfer(address to, uint amount) public returns (bool success) {
        balances.send(msg.sender, to, amount);
        emit Transfer(msg.sender, to, amount);
        return true;

    }

    function transferFrom(address from, address to, uint amount) public returns (bool success) {
        require(allowed[from][msg.sender] >= amount);
        allowed[from][msg.sender] -= amount;
        balances.send(from, to, amount);
        emit Transfer(from, to, amount);
        return true;
    }

    function approve(address spender, uint tokens) public returns (bool success) {
        require(allowed[msg.sender][spender] == 0, "");
        allowed[msg.sender][spender] = tokens;
        emit Approval(msg.sender, spender, tokens);
        return true;
    }
}
```

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

Closes https://github.com/ethereum/solidity/issues/4636

I wonder if the example needs more explanation? Nicely this also properly closes an FAQ point that we closed a while back that related to checking balances.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

**Comment by leonardoalt** [comment]
I agree with https://github.com/ethereum/solidity/issues/4636#issuecomment-409585998 to use `move` instead of `send`, because of the builtin `send`.

**Comment by ChrisChinchilla** [comment]
2 vs 3 on this, so updating :)

File: docs/examples/modular.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            balances.move(msg.sender, to, amount);
+            emit Transfer(msg.sender, to, amount);
+            return true;
+
+        }
+
+        function transferFrom(address from, address to, uint amount) public returns (bool success) {
+            require(allowed[from][msg.sender] >= amount);
+            allowed[from][msg.sender] -= amount;
+            balances.send(from, to, amount);
```
Comment (leonardoalt) [inline docs/examples/modular.rst]:
```suggestion
            balances.move(from, to, amount);
```

**Review by leonardoalt — CHANGES_REQUESTED** [review]

**Review by leonardoalt — APPROVED** [review]

**Review by leonardoalt — CHANGES_REQUESTED** [review]
Please rebase

**Comment by ChrisChinchilla** [comment]
@leonardoalt rebased and squashed

File: docs/examples/modular.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.. index:: contract;modular, modular contract
+
+*****************
+Modular Contracts
+*****************
+
+A modular approach to building your contracts helps you prevent overflow risks
+and unexpected tokens. In the example below, the contract uses the ``send`` method
```
Comment (leonardoalt) [inline docs/examples/modular.rst]:
```suggestion
and unexpected tokens. In the example below, the contract uses the ``move`` method
```

File: docs/examples/modular.rst (hunk 1 above)
Comment (leonardoalt) [inline docs/examples/modular.rst]:
```suggestion
            balances.move(from, to, amount);
```

**Review by leonardoalt — CHANGES_REQUESTED** [review]

**Comment by ChrisChinchilla** [comment]
@leonardoalt I don't know why the changes keep undoing, but looks good.

**Comment by leonardoalt** [comment]
Can you squash again? Sorry...

**Comment by ChrisChinchilla** [comment]
Done @leonardoalt

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**